mistletoe/block_token.py

```diff
@@ -324,14 +324,14 @@ def read(cls, lines):
                 and not Quote.start(next_line)):
 
             # check if next_line starts List
-            list_pair = ListItem.parse_marker(next_line)
+            marker_tuple = ListItem.parse_marker(next_line)
             if (len(next_line) - len(next_line.lstrip()) < 4
-                    and list_pair is not None):
-                prepend, leader = list_pair
-                # non-empty list item
-                if next_line[:prepend].endswith(' '):
```

Comment:
This is one place in the code where tabs were not handled correctly, yes. The original code worked for spaces but not for tabs just as you write.

And yes, there was some support for tab expansion but it didn't work according to the spec: "in contexts where spaces help to define block structure, tabs behave as if they were replaced by spaces with a tab stop of 4 characters." Tabs were always replaced with four spaces, when they should have been replaced with between 1 and 4 spaces depending on the position on the line. There is a tab stop every four positions on the line, and that's what the tabs should expand to.

The tab expansion added now is not perfect, because it doesn't take into account leading space that has been cut off from for example a list item. But it works as long as tabs are applied consistently, i.e., not mixing tabs and spaces too much. The "perfect" implementation is still possible but it's probably quite some work.

I see your point about how examples 312 and 313 are mentioned in the commit message to make it sound like they're related to tab expansion. They are *not*, but they *are* related to leading space in list items and  therefore they happened to get fixed by the same change. I'll try and make that a bit more clear.

Reply:
Thank you for the explanation. I have tried to reformulate the commit message once again - see my next comment.